src/lib/concurrency/transaction_manager.cpp

```diff
@@ -23,7 +24,40 @@ TransactionManager::TransactionManager()
 CommitID TransactionManager::last_commit_id() const { return _last_commit_id; }
 
 std::shared_ptr<TransactionContext> TransactionManager::new_transaction_context() {
-  return std::make_shared<TransactionContext>(_next_transaction_id++, _last_commit_id);
+  const TransactionID snapshot_commit_id = _last_commit_id;
+  return std::make_shared<TransactionContext>(_next_transaction_id++, snapshot_commit_id);
+}
+
+void TransactionManager::register_transaction(const CommitID snapshot_commit_id) {
+  std::unique_lock<std::mutex> lock(_mutex_active_snapshot_commit_ids);
+  _active_snapshot_commit_ids.insert(snapshot_commit_id);
+}
+
+void TransactionManager::deregister_transaction(const CommitID snapshot_commit_id) {
+  std::unique_lock<std::mutex> lock(_mutex_active_snapshot_commit_ids);
+
+  auto it =
+      std::find(std::begin(_active_snapshot_commit_ids), std::end(_active_snapshot_commit_ids), snapshot_commit_id);
```

Comment:
I know it is more generic the was you did it, but the rest of the code base would do `_active_snapshot_commit_ids.begin()` over `std::begin(...)`, so I'd opt for the former for consistency reasons. Same in the `if` below.